### PR TITLE
Update colorbar, inset, legend and logo baseline images for GMT 6.2.0

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

Fix five broken tests on GMT 6.2.0 for fig.colorbar, fig.inset, fig.legend and fig.logo. Differences are mainly due to a different pen thickness setting. Also marking a `test_plot3d_matrix_color` as xfail on Windows due to incorrect `-i` parameter parsing.

Addresses #1320

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/baseline/test_plot3d_matrix_color.png.dvc [hunk 1]
```diff
@@ -1,4 +1,4 @@
 outs:
-- md5: 9b1e94da314923794dca11d20ae3e5df
+- md5: d08eaf4bf196c8680a3b7c0d8f4259a7
   size: 24071
```
Comment (weiji14) [inline pygmt/tests/baseline/test_plot3d_matrix_color.png.dvc:3]: `test_plot3d_matrix_color` still failing on Windows. See https://github.com/GenericMappingTools/pygmt/pull/1322/checks?check_run_id=2756778343#step:11:486:

```
__________________________ test_plot3d_matrix_color ___________________________
Error: Image dimensions did not match.
  Expected shape: (1040, 1729)
    D:\a\pygmt\pygmt\tmp-test-dir-with-unique-name\results\pygmt.tests.test_plot3d.test_plot3d_matrix_color\baseline.png
  Actual shape: (991, 1729)
    D:\a\pygmt\pygmt\tmp-test-dir-with-unique-name\results\pygmt.tests.test_plot3d.test_plot3d_matrix_color\result.png
```

| baseline (correct) | result (wrong) |
|--|--|
| baseline | result |

Comment (seisman) [reply pygmt/tests/baseline/test_plot3d_matrix_color.png.dvc:3]: But it passes on Linux and macOS. I think we can mark it as xfail on Windows, and see if @joa-quim can help debug the Windows-only issue.

**Review by seisman — APPROVED** [review]

File: pygmt/tests/baseline/test_plot3d_matrix_color.png.dvc (hunk 1 above)
Comment (weiji14) [reply pygmt/tests/baseline/test_plot3d_matrix_color.png.dvc]: I've marked `test_plot3d_matrix_color` as xfail on Windows (and reverted the dvc baseline image change to that test) at 327013474302b2c08fe4cb0756d70f91d221920c.

**Comment by joa-quim** [comment]
Sorry, what is it that you want me to test?

**Comment by seisman** [comment]
> Sorry, what is it that you want me to test?

Here is a PyGMT script:
```python
import numpy as np
import pygmt

datafile = "points.txt"

# array
data = np.loadtxt(datafile)
fig = pygmt.Figure()
fig.plot3d(
    data=data,
    zscale=5,
    perspective=[225, 30],
    region=[10, 70, -5, 10, 0, 1],
    projection="X10c",
    style="c0.5c",
    cmap="rainbow",
    frame=["a", "za"],
    i=[0, 1, 2, 2],
)
fig.savefig("plot3d-array.png")
```
The data file is [points.txt](https://github.com/GenericMappingTools/pygmt/files/6605110/points.txt).

The PyGMT test shows that the above script produces the correct (left side) image on Linux/macOS, but produces the wrong (right-side) image on Windows.
| baseline (correct) | result (wrong) |
|--|--|
| baseline | result |

**Comment by joa-quim** [comment]
Ok, but I want to first reproduce this example in Julia and I'm having some errors that I need to debug first.
Anyway I think that in Julia I never use the numeric values directly but parse them and construct a **-i** string

And I'll need some remembering on how to update my PyGMT. Remember that I use the code from repo directly and my GMT win build.

**Pull request merged**